### DISCUSSION TITLE
Publish To Docker Example

### DISCUSSION THREAD

**Discussion by wisamidris7** [microsoft/aspire#2645] [General]
I Have Many Servers And They Are Docker I Don't Want To Use Azure Cause I Already Have Docker Servers
Does Aspire Has Example Or Anything Make Publishing Aspire Project To Docker Or Docker Compose Generator
And Thanks!

**Comment by davidfowl** [comment]
You can use [Aspir8](https://github.com/prom3theu5/aspirational-manifests) to deploy to docker compose and Kubernetes


https://medium.com/@josephsims1/aspire-aspi8-deploy-microservices-effortlessly-with-cli-no-docker-or-yaml-needed-f30b58443107

**Reply by ManrajPatel** [reply]
I have published .net aspire app to local Kubernetes cluster using Aspire8, by following the steps depicted in the provided medium link. 
I have docker desktop on my windows system. 

Everything worked successfully and I can see the containers in docker desktop app.
But I do not see port mappings under Posrt(s) column and hence cannot access the aspire web app into web browser.

@davidfowl Can you please help here? Are any steps missing in the provided medium article?

![image](https://github.com/user-attachments/assets/6b392a99-0218-431d-be5f-0cd862e6a08c)

**Comment by wisamidris7** [comment]
Is This A Tool Provide publishing docker

**Comment by wisamidris7** [comment]
i mean to be like support or command to be easliy publish your docker aspire application

**Reply by wisamidris7** [reply]
okay that awesome but there are problem how to

https://github.com/dotnet/aspire/discussions/2645#discussioncomment-8682088

**Comment by wisamidris7** [comment]
and in publish how i'm gonna pass enviroment varibles into webassembly applications

**Reply by wisamidris7** [reply]
cause blazor webassembly does not have any access to anything that docker pass only appsettings.json put in the wwwroot

**Reply by davidfowl** [reply]
Passing environment variables doesn't work for blazor webassembly applications

**Reply by wisamidris7** [reply]
okay what i do for passing uris and things like this

is it the only way to do is to write it in appsettings.json

**Reply by wisamidris7** [reply]
and what i trying to do is the http client reslover cause i have clients or services in serviceDefaults uses like this

http://my-app
as BaseAddress

and there no client with name my-app passed to blazor webassembly cause it can't access enviroment

**Reply by wisamidris7** [reply]
i maked it like this for now until you found soultion for this

![image](https://github.com/dotnet/aspire/assets/104096256/538b90b7-bcc6-4827-b2e6-f8acf58df26f)
 
in appsettings.json

**Comment by wisamidris7** [comment]
i mean blazor webassembly application

**Comment by wisamidris7** [comment]
does this has been fixed in aspire or it's impossible to be done

more information in here
https://github.com/dotnet/aspire/discussions/2645#discussioncomment-8692902
